sale_layout_category_hide_detail/views/sale_views.xml

```diff
@@ -0,0 +1,71 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2019 Tecnativa - Ernesto Tejeda
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <record id="view_order_form" model="ir.ui.view">
+        <field name="name">sale.order.hide.detail.form</field>
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='order_line']/form//field[@name='name']" position="after">
+                <group>
+                    <field name="show_details"
+                           attrs="{'invisible': [('display_type', '!=', 'line_section')]}"/>
+                    <field name="show_subtotal"
+                           attrs="{'invisible': ['|', ('display_type', '!=', 'line_section'), ('show_details', '!=', True)]}"/>
+                </group>
+            </xpath>
+            <xpath expr="//field[@name='order_line']/tree" position="inside">
+                <field name="show_details"
+                       widget="boolean_fa_icon"
+                       options="{
+                            'show_in_line_section': true,
+                            'fa_icons': {
+                                'icon_true': 'fa-eye',
+                                'icon_false': 'fa-eye-slash',
+                            },
+                            'terminology': {
+                                'hover_true': 'Switch to: details hidden',
+                                'hover_false': 'Switch to: details shown',
+                            }
+                       }"
+                       attrs="{'invisible': [('display_type', '!=', 'line_section')]}"/>
+                <field name="show_subtotal"
+                       widget="boolean_fa_icon"
```

Comment:
I have some questions about this new widget.

1. Have you considered using `boolean_toggle` widget? I think it should be enough for your use case: 
![Peek 09-07-2019 12-58](https://user-images.githubusercontent.com/973709/60886334-98b7b780-a249-11e9-8079-3e07ed68a769.gif)

   The fact that it's not an eye doesn't seem to impact much in user experience IMHO.

1. In case it's not enough, better call it `boolean_icon`. In the future we might use a different icon set, so it's better to keep things unbranded if possible.

Reply:
About the `boolean_toggle`, I prefer this icon set without column name, as it would require a lot of width space. That's the main reason for this. Maybe we can extract the widget in OCA/web...

Reply:
OK, I just did a quick test and this is how it looks:

![Captura de pantalla de 2019-07-09 13-41-01](https://user-images.githubusercontent.com/973709/60888645-382b7900-a24f-11e9-8eb3-659a856a4b89.png)

I can see the difference, indeed.

Then we need to handle:

1. Translations. Important!
1. Extract into OCA/web.

Reply:
Let's put this in the ROADMAP as we don't need this widget for other things and this task has required too much time, so we shouldn't increase it with this extraction.

Reply:
OK, but at least the translations issue should be fixed IMHO. Easier to do if not extracting the widget, BTW.

Reply:
OK, seems a good solution.